wgengine/monitor/monitor_windows.go

```diff
@@ -25,94 +24,67 @@ type eventMessage struct {
 func (eventMessage) ignore() bool { return false }
 
 type winMon struct {
-	logf   logger.Logf
-	ctx    context.Context
-	cancel context.CancelFunc
-
-	mu                    sync.Mutex // guards following
+	logf                  logger.Logf
+	ctx                   context.Context
+	cancel                context.CancelFunc
+	messagec              chan eventMessage
 	addressChangeCallback *winipcfg.UnicastAddressChangeCallback
 	routeChangeCallback   *winipcfg.RouteChangeCallback
-	messagec              chan eventMessage
 }
 
 func newOSMon(logf logger.Logf, _ *Mon) (osMon, error) {
-	m := &winMon{
-		logf: logf,
-	}
+	m := &winMon{logf: logf}
 
-	cba, err := winipcfg.RegisterUnicastAddressChangeCallback(m.unicastAddressChanged)
+	var err error
+	m.addressChangeCallback, err = winipcfg.RegisterUnicastAddressChangeCallback(m.unicastAddressChanged)
 	if err != nil {
 		m.logf("winipcfg.RegisterUnicastAddressChangeCallback error: %v", err)
 		return nil, err
 	}
 
-	cbr, err := winipcfg.RegisterRouteChangeCallback(m.routeChanged)
+	m.routeChangeCallback, err = winipcfg.RegisterRouteChangeCallback(m.routeChanged)
 	if err != nil {
-		cba.Unregister()
+		m.addressChangeCallback.Unregister()
 		m.logf("winipcfg.RegisterRouteChangeCallback error: %v", err)
 		return nil, err
 	}
 
-	m.addressChangeCallback = cba
-	m.routeChangeCallback = cbr
-
 	m.ctx, m.cancel = context.WithCancel(context.Background())
+	m.messagec = make(chan eventMessage, 1)
```

Comment:
move this up to the top, like

```go
m := &winMon{
		logf: logf,
		messagec: make(chan eventMessage, 1),
	}
```

Reply:
OK, will do. It's again my bug to shave nano seconds by avoiding doing things (create chan in this case) unless I'm sure that it's needed. And in this code chan will not be used if error happens and I return `nil`, `err` few lines below... Just explaining. Will change.